### PR TITLE
[WFLY-10531] Wildfly leaks ActiveMQ connections

### PR DESCRIPTION
**Pull request by JiriOndrusek**
Issue: https://issues.jboss.org/browse/WFLY-10531

Fix of regression caused by https://github.com/wildfly/wildfly/pull/10614

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:
- [ ] Pull Request title is properly formatted: `[WFLY-XYZ] Subject` or `WFLY-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue(s)
- [ ] Pull Request contains description of the issue(s)
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted

For bigger changes, major and minor component upgrades make sure your PR also meets following requirements:
- [ ] Pull Request requires a change to the documentation
- [ ] Documentation have been updated accordingly
- [ ] Tests were added to cover changes

For new features ensure as well:
- [ ] Analysis was done
- [ ] Test Plan has been done
- [ ] Tests were verified in advance

If you are not an active contributor of the WildFly project you can request sponsorship by one of the members to help guide you through the process.

### CODE REVIEW

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/deployment/injection/AbstractJMSContext.java [hunk 1]
```diff
@@ -65,11 +70,21 @@ private JMSContext createContext(JMSInfo info, ConnectionFactory connectionFacto
         return context;
     }
 
+    /**
+     * If context is closed by transaction listener, it should not be closed again in PostDestroy method.
```
Comment (jmesnil) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/deployment/injection/AbstractJMSContext.java]: I suppose you mean in `@PreDestroy`, not `@PostDestroy`?

Comment (JiriOndrusek) [reply messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/deployment/injection/AbstractJMSContext.java]: @jmesnil  Yes, my bad, I'll fix it in a moment

**Review by jwgmeligmeyling — APPROVED** [review]
Still have no clue why this issue exists in the first place, but the fix is obvious (falling back to the old behaviour if the new behaviour fails to close the JMS context) and with this patch applied, the issue is resolved for me.

**Comment by JiriOndrusek** [comment]
Superseded by https://github.com/wildfly/wildfly/pull/11715

**Pull request closed without merging**